### PR TITLE
Disable `Next` button unless source URL is valid

### LINKED ISSUE #2388: Source repository input does not validate value as URL
When creating from a source repo in the web console, I can type anything into the `Source repository` field and click next. We make no attempt to validate the URL. Using `<input type="url">` would fix this.

### PR DESCRIPTION
**Pull request by spadgett**
Fixes #2388

openshift_management_console

### CODE REVIEW

**Comment by spadgett** [comment]
@liggitt let me know what you think. Input `type="url"` doesn't work because it doesn't like `git:` or `ssh://`. I tried to be conservative with the regex.

**Comment by liggitt** [comment]
can you run the validation in javascript? I'd much rather do this:
```
var u = URI(value);
return u.is("absolute") && u.is("url");
```

I'd rather just check basic things on that page

**Comment by spadgett** [comment]
@liggitt Will `is('url')` reject things like `git@github.com:user/project.git`. I assume that's valid.

**Comment by liggitt** [comment]
hmm... looks like it doesn't. I really don't want to get in the business of thorough url validation here

**Comment by spadgett** [comment]
@liggitt Yeah, I understand... I'd say we either do very basic regex validation -- I've tried to keep it simple -- or just stick to `required` only and let it fail later.

FWIW, here's the regex:

debuggex__online_visual_regex_tester__javascript__python__and_pcre_

**Comment by liggitt** [comment]
yeah, even that misses things like `git+ssh`...

File: assets/app/views/create.html [hunk 1]
```diff
@@ -4,25 +4,30 @@
...
+          <form name="from_source_form" class="form-inline gutter-bottom">
+            <h2>Source repository</h2>
+            <div class="row">
+              <div class="col-md-8">
+                <div class="input-group">
+                  <span ng-class="{'has-error': from_source_form.$invalid && from_source_form.from_source_url.$touched}">
+                    <input class="form-control input-lg"
+                      name="from_source_url"
+                      placeholder="URL for source repository"
+                      pattern="(((ssh|http(s)?)://)|(git(@[\w\.]+)?:)).+"
```
Comment (liggitt) [inline assets/app/views/create.html]: could go super-simple and do `(ssh://|http://|https://|git).+`...
Is there any indication there's an invalid value?
Is that case-sensitive?

**Comment by spadgett** [comment]
@liggitt OK, let me take out the regex, but I'll leave `Next` disable until something is typed at least.

File: assets/app/views/create.html (hunk 1 above)
Comment (spadgett) [inline assets/app/views/create.html]: There's an indication of an invalid value when you blur the field.
It is case sensitive.

**Comment by spadgett** [comment]
@liggitt OK, an even simpler regex. Let me know what you think.

File: assets/app/scripts/controllers/create.js [hunk 2]
```diff
@@ -14,8 +14,6 @@ angular.module('openshiftConsole')
 
     $scope.templatesByTag = {};
 
-    $scope.sourceURLPattern = /^((ftp|http|https|git):\/\/(\w+:{0,1}\w*@)|git@)?([^\s@]+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-\/]))?$/;
```
Comment (liggitt) [inline assets/app/scripts/controllers/create.js]: keep the pattern in the controller... it's used by `assets/test/spec/controllers/create.js`

**Comment by spadgett** [comment]
@liggitt one more look... Now using `ng-pattern="sourceURLPattern"`

**Comment by liggitt** [comment]
[merge]

**Pull request merged**